geps/gep-1742.md

```diff
@@ -301,13 +301,168 @@ Could not find any HTTP specific timeouts. PRs welcomed. 😊
+    This field is optional `Extended` support. Typically used in conjuction with retry configuration,
+    if supported by an implementation.
+    Note that retry configuration will be the subject of a separate GEP (GEP-1731).
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant P as Proxy
+    participant U as Upstream
+    C->>P: Connection Started
+    note left of P: timeouts.request start time (min)
+    C->>P: Starts sending Request
+    C->>P: Finishes Headers
+    C->>P: Finishes request
+    note left of P: timeouts.request start time (max)
+    P->>U: Connection Started
+    note right of P: timeouts.backendRequest start time
+    P->>U: Starts sending Request
+    P->>U: Finishes request
+    P->>U: Finishes Headers
+    U->>P: Starts Response
+    U->>P: Finishes Headers
+    note right of P: timeouts.backendRequest end time
+    note left of P: timeouts.request end time
+    U->>P: Finishes Response
+    note right of P: Repeat if retry
+    P->>C: Starts Response
+    P->>C: Finishes Headers
+    P->>C: Finishes Response
+    Note right of P: Repeat if connection sharing
+    U->>C: Connection ended
+```
 
+Both timeout fields are string duration values as specified by
+[Golang time.ParseDuration](https://pkg.go.dev/time#ParseDuration) and MUST be >= 1ms.
 
 ## Alternatives
 
-(List other design alternatives and why we did not go in that
-direction)
+Timeouts could be configured using policy attachments or in objects other than `HTTPRouteRule`.
+
+### Policy Attachment
+
+Instead of configuring timeouts directly on an API object, they could be configured using policy
+attachments. The advanatage to this approach would be that timeout policies can not only be
+configured for an `HTTPRouteRule`, but can also be added/overriden at a more fine
+(e.g., `HTTPBackendRef`) or course (e.g. `HTTPRoute`) level of granularity.
+
+The downside, however, is complexity introduced for the most common use case, adding a simple
+timeout for an HTTP request. Setting a single field in the route rule, instead of needing to
+create a policy resource, for this simple case seems much better.
+
+Note that in the future, we could use policy attachments to configure other kinds of less common
+timeouts that may be needed. The default values of the proposed timeout fields could also be overriden
+using policy attachments in the future. For example, a policy attachment could be used to set the
+default value of `rules.timeouts.request` for all routes under an `HTTPRoute` or `Gateway`.
+
+### Other API Objects
+
+The new timeouts field could be added to a different API struct, instead of `HTTPRouteRule`.
+
+Putting it on an `HTTPBackendRef`, for example, would allow users to set different timeouts for different
+backends. This is a feature that we believe has not been requested by existing proxy or service mesh
+clients and is also not easily implementable.
```

Comment:
I would omit the comment about "not easily implementable" unless you explain why.  Maybe also add this to Non-Goals?
```suggestion
clients and is not being considered for this proposal.
```

Reply:
Clarified not easily implementable